pymaker/auctions.py

```diff
@@ -155,19 +159,39 @@ def tick(self, id: int) -> Transact:
 
         return Transact(self, self.web3, self.abi, self.address, self._contract, 'tick', [id])
 
-    def get_past_lognotes(self, number_of_past_blocks: int, abi: list) -> List[LogNote]:
-        assert isinstance(number_of_past_blocks, int)
+    def get_past_lognotes(self, abi: list, from_block: int, to_block: int = None, chunk_size=20000) -> List[LogNote]:
+        current_block = self._contract.web3.eth.blockNumber
+        assert isinstance(from_block, int)
+        assert from_block < current_block
+        if to_block is None:
+            to_block = current_block
+        else:
+            assert isinstance(to_block, int)
+            assert to_block >= from_block
+            assert to_block <= current_block
+        assert chunk_size > 0
         assert isinstance(abi, list)
 
-        block_number = self._contract.web3.eth.blockNumber
-        filter_params = {
-            'address': self.address.address,
-            'fromBlock': max(block_number - number_of_past_blocks, 0),
-            'toBlock': block_number
-        }
+        start = from_block
+        end = None
```

Comment:
Not sure what the intended use case is for `get_past_lognotes` (or `past_logs` in below subclasses), but it may be worth considering adding this statement on line 174, which I took from `pymaker.dss.Vat.past_frobs()`: 

`logger.debug(f"Consumer requested log data from block {from_block} to {to_block}")`

Reply:
Done.